### PR TITLE
Add JWTVerifer added isValid method

### PR DESCRIPTION
**Pull request by smojum**
JWTVerifer added isValid method. Reason for the addition
1) it shows false exception for instrumented tools like Dynatrace
2) Functional programming without exception

### References

Please include relevant links supporting this change such as a:
#354 https://github.com/auth0/java-jwt/issues/354

### Testing

Junit tests have been added. Test coverage has been verified.
- [X] This change adds test coverage
- [X] This change has been tested on the latest version of Java or why not

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [X] All existing and new tests complete without errors

### CODE REVIEW

File: lib/src/main/java/com/auth0/jwt/JWTVerifier.java [hunk 1]
```diff
@@ -388,6 +388,26 @@ public DecodedJWT verify(DecodedJWT jwt) throws JWTVerificationException {
...
+        try {
+            verify(token);
+            return Boolean.TRUE;
+        } catch (Exception e) {
+            return Boolean.FALSE;
+        }
+    }
+
+    @Override
+    public Boolean isValid(DecodedJWT jwt) {
```
Comment (joshcanhelp) [inline lib/src/main/java/com/auth0/jwt/JWTVerifier.java]: When would a decoded JWT be used? As in, why would someone pass in an already decoded token here? What would the `verify()` function check?

Comment (smojum) [reply lib/src/main/java/com/auth0/jwt/JWTVerifier.java]: the reason I added the isValid(DecodedJWT jwt) just to keep it consistent with verify methods. As in verify method take either token or DecodedJWT.

Comment (smojum) [reply lib/src/main/java/com/auth0/jwt/JWTVerifier.java]: I may have misunderstood your question. the reason behind to add this method is to 
1) the equivalent method verify throws JWTException which is not very functional programming friendly
2) when Exception is thrown some monitoring tools which does the instrumentation of code to see for any exceptions are showing a lot of JWT exception, when the intent in the code is just to verify whether it is a valid JWT token.

**Comment by smojum** [comment]
Hi, 

Just checking, is there anything else needed from me.

**Comment by lbalmaceda** [comment]
Not needed. Left a comment on the linked issue to discuss.

**Comment by lbalmaceda** [comment]
Closing since it doesn't align with how we want this library to be consumed. The reason the verification fail should be relevant to the developer, not ignored.

**Pull request closed without merging**